pkg/app/oauth2/oauth2.go

```diff
@@ -35,15 +36,37 @@ const (
 	GoogleOAuth2Provider = "Google"
 )
 
+const (
+	JWTAuthType     = "jwt"
+	OAuthClientType = "oauth"
```

Comment:
What about introduce a new OAuthType like other managers(e.g. InstanceManager, AccountManager, SecretManager) do?

Reply:
Good suggestion, this should make the code more readable, but an open question here is how should we classify these auth types?

What @chyijia tries to add here is the JWT authentication flow, which does not belong to the OAuth protocol if I understand correctly. Also, CO currently uses `Provider` to decide which `oauth2Helper` should be created (https://github.com/google/cloud-android-orchestration/blob/main/cmd/cloud_orchestrator/main.go#L100). However, in our case, I think this should be based on which authentication protocol is using. So how about defining two different AuthHelpers, one is `GoogleOAuth2Helper` and another is `JWTHelper`?

We may need @ser-io's suggestions about this small refactor.

Reply:
To make it clear, I mean something like:
```
switch config.AccountManager.Auth.Type {
case JWT:
	helper = NewJWTHelper(...)
case OAuth2:
	switch config.AccountManager.Auth.OAuth2.Provider {
	case appOAuth2.GoogleOAuth2Provider:
		helper = NewGoogleOAuth2Helper(...)
	default:
		...
	}
default:
	...
}
WDYT?

Reply:
Ah, I got a misunderstanding about JWT. It wasn't a part of OAuth protocol. Sorry for giving confusion.

I think having 2 different AuthHelper is fine. And not sure about listing type inside of `AccountManager`, but no strong opinion from my side. Your suggestion SGTM at least, but prefer to listen @ser-io. He might have a different view.